src/api-service/__app__/onefuzzlib/azure/subnet.py

```diff
@@ -16,21 +17,41 @@
 from .network_mgmt_client import get_network_client
 
 
-def get_subnet_id(resource_group: str, name: str, subnet_name: str) -> Optional[str]:
+def get_vnet(resource_group: str, name: str) -> Optional[VirtualNetwork]:
     network_client = get_network_client()
     try:
-        subnet = network_client.subnets.get(resource_group, name, subnet_name)
-        return cast(str, subnet.id)
+        vnet = network_client.virtual_networks.get(resource_group, name)
+        return cast(VirtualNetwork, vnet)
     except (CloudError, ResourceNotFoundError):
         logging.info(
-            "subnet missing: resource group:%s name:%s subnet_name:%s",
+            "vnet missing: resource group:%s name:%s",
             resource_group,
             name,
-            subnet_name,
         )
     return None
 
 
+def get_subnet(
+    resource_group: str, vnet_name: str, subnet_name: str
+) -> Optional[Subnet]:
+    # Has to get using vnet. That way NSG field is properly set up in subnet
+    vnet = get_vnet(resource_group, vnet_name)
+    if vnet:
+        for subnet in vnet.subnets:
+            if subnet.name == subnet_name:
+                return subnet
+
+    return None
+
+
+def get_subnet_id(resource_group: str, name: str, subnet_name: str) -> Optional[str]:
+    subnet = get_subnet(resource_group, name, subnet_name)
+    if subnet:
+        return cast(str, subnet.id)
```

Comment:
Instead of a `cast`, can gate this on a check to `isinstance(subnet.id, str)`? That should convince `mypy` and also actually ensure the type is what we think it is.